src/core/meshcop/leader.cpp

```diff
@@ -86,8 +86,7 @@ void Leader::HandlePetition(Coap::Header &aHeader, Message &aMessage, const Ip6:
     data.mCommissionerSessionId.SetCommissionerSessionId(++mSessionId);
 
     data.mSteeringData.Init();
-    data.mSteeringData.SetLength(1);
-    data.mSteeringData.Clear();
+    data.mSteeringData.SetLength(0);
```

Comment:
Section 8.4.2.2.1 states the following:

> Steering Data: indicates which Joiners should attempt join the network. A Bloom Filter with zero value indicates that joining is disabled, which is the default for a new Commissioner.

As a result, the Commissioner is responsible for setting the Steering Data to allow Joiners onto the network.